crates/autopilot/src/domain/eth/mod.rs

```diff
@@ -45,6 +45,19 @@ impl TokenAmount {
     }
 }
 
+/// An ERC20 sell token amount.
+///
+/// https://eips.ethereum.org/EIPS/eip-20
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, From, Into)]
```

Comment:
shouldn't we have a crate containing all these types? they seem to repeat themselves over the codebase :thinking:

Reply:
Would the `model` crate be an appropriate place for those?

Reply:
No, those types are not models, they are domain types. I wouldn't place them in the `model` crate, since `model` implies the types implement `serde` (which is not the case)

Reply:
Would you want to share these types across driver and autopilot? I'm not sure those should have a coupled domain (it's not so much work to re-define them and I'm worried we will start leaking things that would be driver specific into a shared crate).

Although there is an argument to be made that their may be some over-arching COW domain (which includes types around tokens, trades, etc). But frankly this is how the model crate came along which has matured pretty poorly imo.

Reply:
@fleupold (also cc @sunce86 for feedback), my idea is to have a crate called `cow-types` which contains very basic type which are 100 % common to every domain. For example:

1. type `Address` defined exactly the same in three different places:
- [Autopilot](https://github.com/cowprotocol/services/blob/main/crates/autopilot/src/domain/eth/mod.rs#L6)
- [Driver](https://github.com/cowprotocol/services/blob/main/crates/driver/src/domain/eth/mod.rs#L111)
- [Solvers](https://github.com/cowprotocol/services/blob/main/crates/solvers/src/domain/eth/mod.rs#L90)

Basically majority of the `eth` module is copied all over at least those three crates. The idea of `cow-types` is to place "absolute truth" types, such as `Asset`, `SellTokenAmount`, `Token`, etc. I don't want/think placing in such crate domain types such as `Auction` is a good idea, because `Auction` is no absolute truth for every domain, since each domain may have their own implementation/definition of it.

Reply:
I would like to also make those types not to leak their internal type (e.g. in case of the `Address` type, its `H160` type). Derive `From/Into` with the `derive_more` crate I introduced into the workspace crate, and do `H160::from()` or `variable.into()` for getting its `H160` instead of `.0`. So it would behave like a "proper type crate".

Reply:
I'd prefer to keep them separated. They are part of their `domain` modules and probably shouldn't be shared across multiple crates. We can't guarantee they will always stay same for all crates.